src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -343,28 +343,29 @@ public void TestCreateNoWindowProperty(bool value)
         }
 
         [Fact]
-        public void TestWorkingDirectoryProperty()
+        public void TestWorkingDirectoryPropertyInChildProcess()
         {
             CreateDefaultProcess();
 
             // check defaults
             Assert.Equal(string.Empty, _process.StartInfo.WorkingDirectory);
 
-            Process p = CreateProcessLong();
-            p.StartInfo.WorkingDirectory = Directory.GetCurrentDirectory();
+            ProcessStartInfo psi = new ProcessStartInfo();
+            string sourcePath = Directory.GetCurrentDirectory();
 
-            try
-            {
-                p.Start();
-                Assert.Equal(Directory.GetCurrentDirectory(), p.StartInfo.WorkingDirectory);
-            }
-            finally
+            foreach (string newPath in Directory.GetFiles(sourcePath, "*.*", SearchOption.AllDirectories))
+                File.Copy(newPath, newPath.Replace(sourcePath, TestDirectory), true);
+
+            psi.WorkingDirectory = TestDirectory;
```

Comment:
Is TestDirectory definitely different to CurrentDirectory() ? If it isn't, this test is not going to do anything.

Set WorkingDirectory to Environment.SystemDirectory (assuming that always exists on Unix)
Also to be sure I would Assert that WorkingDirectory != CurrentDirectory.

Reply:
The test Directory is always different from Current Directory 
```TestDirectory = Path.Combine(Path.GetTempPath(), GetType().Name + "_" + Path.GetRandomFileName());```
why should we set WorkingDirectory to Environment.SystemDirectory  ?

Reply:
Personally I would do it as it makes that obvious. However it's fine  to leave as you point out.

Reply:
i will add a ```Assert.NotEqual(TestDirectory, Directory.getcurrentdirectory)``` to make it clear that they should never be equal